frame/utility/src/lib.rs

```diff
@@ -376,12 +394,13 @@ decl_module! {
 						approvals: vec![who.clone()],
 					});
 					Self::deposit_event(RawEvent::NewMultisig(who, id, call_hash));
+					// Call is not made, so we can return that weight
+					return Ok(Some(call.get_dispatch_info().weight).into())
 				} else {
 					return call.dispatch(frame_system::RawOrigin::Signed(id).into())
-						.map(|_| ()).map_err(|e| e.error)
+						.map(|_| None.into()).map_err(|e| e.error.into())
```

Comment:
IMHO you should take actual weight of the call into account here.

Reply:
yes, this is a good catch